Migrate Figure.coast tests to dvc

willschlitzer opened the pull request:
Changing the tests in test_coast.py to use `dvc` and `mpl_image_compare`. An update/fix to #1102.

Related to #1131.


**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

weiji14 commented on pygmt/tests/test_coast.py:
```diff
@@ -14,60 +13,37 @@ def test_coast():
-        S="blue",
-        D="c",
-        A=10000,
+        region="-30/30/-40/40",
+        projection="m0.1i",
```
Good opportunity to change to SI units, and use a Python list for `region`.
```suggestion
        region=[-30, 30, -40, 40],
        projection="m0.25c",
```

weiji14 commented on pygmt/tests/test_coast.py:
```diff
@@ -14,60 +13,37 @@ def test_coast():
 
-@check_figures_equal()
+@pytest.mark.mpl_image_compare
 def test_coast_aliases():
```
This is one of the top 3 slowest tests as reported in #840, can you try and speed it up? E.g. using resolution="c" for crude? Maybe also plot a smaller geographical region.

willschlitzer replied:
Done 2f7ba86 and 0fcbdfc

seisman commented:
The [`coast` wrapper](https://github.com/GenericMappingTools/pygmt/blob/master/pygmt/src/coast.py) is very simple and only has a few lines of codes. 

As documented in the [contributing guides](https://github.com/GenericMappingTools/pygmt/blob/master/CONTRIBUTING.md#testing-your-code),

> The highest priority for tests should be the Python-specific code, such as numpy, pandas, and xarray objects and the virtualfile mechanism.

I feel some of the tests are unnecessary. Tests for a global map, a regional map, a map using `dcw` should be enough.

willschlitzer commented:
> The [`coast` wrapper](https://github.com/GenericMappingTools/pygmt/blob/master/pygmt/src/coast.py) is very simple and only has a few lines of codes. 
> 
> 
> 
> As documented in the [contributing guides](https://github.com/GenericMappingTools/pygmt/blob/master/CONTRIBUTING.md#testing-your-code),
> 
> 
> 
> > The highest priority for tests should be the Python-specific code, such as numpy, pandas, and xarray objects and the virtualfile mechanism.
> 
> 
> 
> I feel some of the tests are unnecessary. Tests for a global map, a regional map, a map using `dcw` should be enough.
> 
> 
> 
> Edit: IMHO, only the following tests are needed:
> 
> 
> 
> - test_coast_iceland
> 
> - test_coast_world_mercator
> 
> - test_coast_required_args
> 
> - test_coast_dcw_single
> 
> - test_coast_dcw_list

Sounds good! I'll shorten it down.

willschlitzer commented:
@seisman @weiji14 I removed the tests that we deemed not needed and deleted the `.dvc` files; do I have to do anything else to fully remove these tests?

seisman commented on pygmt/tests/test_coast.py:
```diff
@@ -1,94 +1,21 @@
+def test_coast_iceland():
     """
-    Simple plot from the GMT docs.
+    Test passing in region as a list.
```
```suggestion
    Test plotting a regional map with coastlines.
```

seisman commented on pygmt/tests/test_coast.py:
```diff
@@ -1,94 +1,21 @@
-
 @pytest.mark.mpl_image_compare
 def test_coast_world_mercator():
     """
```
> Test passing generating a global Mercator map with coastlines.

should be changed to 

> Test generating a global Mercator map with coastlines.

willschlitzer replied:
Updated

seisman commented on pygmt/tests/test_coast.py:
```diff
@@ -1,94 +1,21 @@
-
-
 @pytest.mark.mpl_image_compare
 def test_coast_world_mercator():
```
Please change `projection="M10i",` to "projection="M15c",".

Also change `resolution="l",` to `resolution="c",`.

Both should help decrease the image size.

willschlitzer replied:
Updated 9ad494e

seisman commented on pygmt/tests/test_coast.py:
```diff
@@ -1,94 +1,21 @@
 
 @pytest.mark.mpl_image_compare
-def test_coast():
+def test_coast_iceland():
```
I think we should rewrite this test. Below is the baseline image for this test. It's difficult to verify its correctness. Instead, if we plot a regional map like Japan, Hawaii, or Australia, it's much easier to check.

![image](https://user-images.githubusercontent.com/3974108/112732499-ae36ae80-8f10-11eb-9ef9-ddbbf73674c6.png)

willschlitzer replied:
In 150c45e, I changed the region (had to pick Japan; I miss Okinawa!) and think I made it more readable.

seisman commented on pygmt/tests/baseline/test_coast_iceland.png.dvc:
```diff
@@ -0,0 +1,4 @@
+outs:
```
This file `pygmt/tests/baseline/test_coast_iceland.png.dvc` should be deletete.

seisman replied:
Ping @willschlitzer to address the comment and then this PR should be ready for merge.

willschlitzer replied:
Done in 0a1c5e7
